src/Playwright/Transport/WebSocketTransport.cs

```diff
@@ -0,0 +1,207 @@
+                    {
+                        var offset = DefaultBufferSize * i;
+                        var count = DefaultBufferSize;
+                        var lastMessage = (i + 1) == messagesCount;
+
+                        if ((count * (i + 1)) > messageBuffer.Length)
+                        {
+                            count = messageBuffer.Length - offset;
+                        }
+
+                        await _webSocket.SendAsync(new ArraySegment<byte>(messageBuffer, offset, count), WebSocketMessageType.Text, lastMessage, _readerCancellationSource.Token).ConfigureAwait(false);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Close(ex);
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Close(string closeReason)
+        {
+            if (!IsClosed)
+            {
+                IsClosed = true;
+                _ = _webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, closeReason, _readerCancellationSource.Token).ConfigureAwait(false);
+                TransportClosed?.Invoke(this, new TransportClosedEventArgs { CloseReason = closeReason });
+                _readerCancellationSource?.Cancel();
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Close(Exception ex)
+        {
+            System.Diagnostics.Debug.WriteLine(ex);
+            Close(ex.ToString());
+        }
+
+        private async Task ConnectAsync()
+        {
+            await _webSocket.ConnectAsync(new Uri(_wsEndpoint), _readerCancellationSource.Token).ConfigureAwait(false);
+            ScheduleTransportTask(ReceiveAsync, _readerCancellationSource.Token);
+        }
+
+        private void ScheduleTransportTask(Func<CancellationToken, Task> func, CancellationToken cancellationToken)
+            => Task.Factory.StartNew(() => func(cancellationToken), cancellationToken, TaskCreationOptions.LongRunning, TaskScheduler.Current);
+
+        private async Task ReceiveAsync(CancellationToken token)
+        {
+            try
+            {
+                if (_webSocket.State == WebSocketState.Closed)
+                {
+                    Close("Closed");
+                }
+                else
+                {
+                    byte[] buffer = new byte[DefaultBufferSize];
+
+                    while (!token.IsCancellationRequested && _webSocket.State == WebSocketState.Open)
+                    {
+                        WebSocketReceiveResult result;
+                        using (MemoryStream memoryStream = new())
+                        {
+                            do
+                            {
+                                result = await _webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), _readerCancellationSource.Token).ConfigureAwait(false);
```

Comment:
```suggestion
                                result = await _webSocket.ReceiveAsync(buffer, _readerCancellationSource.Token).ConfigureAwait(false);
```

Reply:
Did you intentionally use `_readerCancellationSource.Token` instead of the `token` you're already provided in the method?

Reply:
Aahh naa my bad this needs to be refactored 👍

Reply:
fixed